src/Sparse/Matrix.jl

```diff
@@ -952,6 +952,162 @@ function Base.insert!(A::SMat{T}, i::Integer, B::SRow{T}) where T
   return A
 end
 
+################################################################################
+#
+#  delete/empty
+#
+################################################################################
+
+@doc Markdown.doc"""
+  delete_row!(A::SMat{T}, i::Int) -> SMat{T}
+
+Deletes $i$-th row of $A$ in place.
+"""
+function delete_row!(A::SMat{T}, i::Int) where T
+  non_zeros = length(A[i].pos)
+  deleteat!(A.rows, i)
+  A.r-=1
+  A.nnz-=non_zeros
+  return A
+end
+
+@doc Markdown.doc"""
+  delete_rows!(A::SMat{T}, I, sorted=true)
+
+Deletes rows in set of indices $I$ of $A$ in place.
+"""
+function delete_rows!(A::SMat{T}, I, sorted=true) where T #elements in I need to be ascending
+  if !sorted
+      sort(I)
+  end
+  for i in length(I):-1:1
+      delete_row!(A, I[i])
+  end
+  return A
+end
+
+@doc Markdown.doc"""
+  delete_zero_rows!(A::SMat{T}, s=1)
+
+Deletes zero rows of $A$ in place.
+"""
+function delete_zero_rows!(A::SMat{T}, s=1) where T #where s denotes the first column where we wanna start
+  for i=A.r:-1:s
+      if A[i].pos == []
+          deleteat!(A.rows, i); A.r-=1
+      end
+  end
+  return A
+end
+
+@doc Markdown.doc"""
+  empty_col!(A::SMat{T}, TA::SMat{T}, j::Int) -> SMat{T}
+
+Deletes non-zero entries in $j$-th column of $A$ in place.
+"""
+function empty_col!(A::SMat{T}, TA::SMat{T}, j::Int) where T #only deletes entries in column j, output same size as input
+  for row in TA[j].pos 
+      i = findfirst(isequal(j), A[row].pos)
```

Comment:
`i` can be nothing (if the column does not occur in this row), then this will crash (and the nnz count is wrong)

Reply:
Also, one can always use `searchsortedfirst` here since the list of columns is (hopefully) sorted, so this is a bit faster :)

Reply:
Has this been fixed? It is also not clear what `TA` is (not mentioned in docstring).

Reply:
I revised some of the docstrings. TA is now mentioned for every function it is used